pkg/common/rotationutil/rotationutil.go

```diff
@@ -5,13 +5,19 @@ import (
 	"math/rand"
 	"time"
 
+	"github.com/sirupsen/logrus"
+
 	"github.com/spiffe/spire/pkg/agent/client"
 )
 
+const (
+	gracePeriodThreshold = 12 * time.Hour
+)
+
 // ShouldRotateX509 determines if a given SVID should be rotated, based
 // on presented current time, and the certificate's expiration.
-func ShouldRotateX509(now time.Time, cert *x509.Certificate) bool {
-	return shouldRotate(now, cert.NotBefore, cert.NotAfter)
+func ShouldRotateX509(now time.Time, cert *x509.Certificate, availabilityTarget time.Duration, log logrus.FieldLogger) bool {
```

Comment:
I'm not sure about this approach,  looks now functions requires to have an availabilityTarged, if it is jwt or x.509, and in both cases we requires to have a logger too.
May we create an struct? and create at the start of the agent? and set the `availabilityTarget` there?  (and maybe add the logger too)
So places where we are using thi public function
somehting like:

```
type RotationUtil interface {
   X509(now time.Time, cert *x509.Certificate) bool{}
   JWTSVIDExpiresSoon(svid *client.JWTSVID, now time.Time) bool {}
}

type rotationUtil struct {
     availabilityTarget  time.Duration
     log logrus.FieldLogger
}

func (r *rotationUtil) X509(now time.Time, cert *x509.Certificate) bool{}
func (r *rotationUtil) JWTSVIDExpiresSoon(svid *client.JWTSVID, now time.Time) bool {}
```

but maybe it is an overkill, what do you think about this? (And I dont like interface name :D)

Reply:
I have refactored some of them and would be happy to review them again.